scripts/set_aws_secrets.py

```diff
@@ -0,0 +1,277 @@
+import argparse
+import json
+from typing import TypedDict
+import os
+import boto3
+
+
+"""
+This script is used to set common secrets in all aws accounts
+This must be passed to this script as gh_auth_token param - eg
+
+poetry run python scripts/set_aws_secrets.py
+"""
+
+
+class Secrets(TypedDict):
+    slack_webhook_url: str
+    splunk_hec_token: str
+    jira_token: str
+    confluence_token: str
+    spine_asid: str
+    spine_ca_chain: str
+    spine_public_certificate: str
+    spine_private_key: str
+    spine_party_key: str
+    eps_signing_cert_chain: str
+    service_search_api_key: str
+    PSU_proxygen_private_key: str
+    PSU_proxygen_public_key: str
+    CPSU_proxygen_private_key: str
+    CPSU_proxygen_public_key: str
+    ClinicalTracker_proxygen_private_key: str
+    ClinicalTracker_proxygen_public_key: str
+    FhirDispensing_proxygen_private_key: str
+    FhirDispensing_proxygen_public_key: str
+    ptl_prescription_signing_public_key: str
+    ptl_prescription_signing_private_key: str
+    PSUNotify_callback_api_key: str
+    PSUNotify_callback_app_name: str
+    PSUNotify_kid_secret: str
+    PSUNotify_private_key_secret: str
+
+
+def get_named_export(all_exports: list, export_name: str, required: bool) -> str | None:
+    export_value = None
+
+    for export in all_exports:
+        if export['Name'] == export_name:
+            export_value = export['Value']
+            break
+
+    if required and export_value is None:
+        raise Exception(f"export {export_name} is required but not found")
+    return export_value
+
+
+def get_all_exports(profile_name: str) -> list:
+    print(f"Getting exports for profile {profile_name}")
+    session = boto3.Session(profile_name=profile_name)
+    # Create a CloudFormation client
+    cf_client = session.client('cloudformation')
+
+    # Get all exports
+    all_exports = []
+    next_token = None
+
+    while True:
+        if next_token:
+            response = cf_client.list_exports(NextToken=next_token)
+        else:
+            response = cf_client.list_exports()
+
+        all_exports.extend(response.get('Exports', []))
+
+        next_token = response.get('NextToken')
+        if not next_token:
+            break
+    return all_exports
+
+
+def get_secret_arns(all_exports: list) -> Secrets:
+    slack_webhook_url = get_named_export(all_exports=all_exports,
```

Comment:
This block is a bit samey - could we store the function args in a list of lists, and process them in a loop?
```
exports = [
    ["key", "export name", <required bool>]
]

def get_secret_arns(all_exports: list) -> Secrets:
    secrets: dict[str, str] = {}
    for key, export_name, required in EXPORTS:
        secrets[key] = get_named_export(all_exports, export_name, required)
    return secrets
```
It'd be a bit easier to add new ones later.